src/components/connection-modal/connection-modal.jsx

```diff
@@ -0,0 +1,36 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+
+import Box from '../box/box.jsx';
+import Modal from '../modal/modal.jsx';
+
+import ScanningStep from '../../containers/scanning-step.jsx';
+import ConnectingStep from './connecting-step.jsx';
+import ConnectedStep from './connected-step.jsx';
+import ErrorStep from './error-step.jsx';
+
+import styles from './connection-modal.css';
+
+const ConnectionModalComponent = props => (
+    <Modal
+        className={styles.modalContent}
+        contentLabel={''}
+        headerClassName={styles.header}
+        onRequestClose={props.onCancel}
+    >
+        <Box className={styles.body}>
+            {props.phase === 'scanning' && <ScanningStep {...props} />}
+            {props.phase === 'connecting' && <ConnectingStep {...props} />}
+            {props.phase === 'connected' && <ConnectedStep {...props} />}
+            {props.phase === 'error' && <ErrorStep {...props} />}
+        </Box>
+    </Modal>
+);
+
+ConnectionModalComponent.propTypes = {
+    onCancel: PropTypes.func.isRequired,
+    phase: PropTypes.string.isRequired,
```

Comment:
It would be useful to document the phases here so that people know which phases there are.  I would suggest setting the phases to constants, set on the component:
```jsx
import keyMirror from 'keymirror';

// keyMirror creates an object with values the same as keys
ConnectionModalComponent.PHASES = keyMirror({
    'scanning': null,
    'connecting': null,
    'connected': null,
    'error': null
});

ConnectionModalComponent.propTypes = {
    phase: PropTypes.oneOf(Object.keys(ConnectionModalComponent.PHASES))
};

/* containers/connection-modal.js */
const PHASES = ConnectionModalComponent.PHASES;

    this.setState({phase: PHASES.scanning})
```